src/lib/yacc/grammar.rs

```diff
@@ -309,7 +323,7 @@ impl YaccGrammar {
 
     /// Return the name of terminal `i` or `None` if it doesn't exist.
     pub fn term_name(&self, i: TIdx) -> Option<&str> {
-        self.term_names.get(usize::from(i)).map_or(None, |x| Some(x))
+        self.term_names.get(usize::from(i)).map_or(None, |x| x.as_ref().map_or(None, |y| Some(&y)))
```

Comment:
It seems odd to me to have `as_ref` and then `Some(&y)` here, is there a simpler formulation for this expression?